server/src/main/java/org/opensearch/search/sandbox/RequestSandboxClassifier.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.sandbox;
+
+import org.opensearch.action.search.SearchRequest;
+
+/**
+ * This class is used to classify co-ordinator search reqyests into  sandboxes
+ */
+public class RequestSandboxClassifier {
+
+    /**
+     *
+     * @param request is a coordinator search request
+     * @return matching sandboxId based on request attributes
+     */
+    public String resolveSandboxFor(final SearchRequest request) {
```

Comment:
This will likely get merged into search pipelines now, Given that we are doing nothing but tagging these requests.

To briefly describe this, We will do the tagging (one time process) of incoming search requests at co-ordinator node and propagate this tagged information to shard level requests as well, this will help us group the requests into these different buckets/sandboxes